src/System.Private.CoreLib/src/System/Threading/ThreadPool.Unix.cs

```diff
@@ -22,103 +22,48 @@ public bool Unregister(WaitHandle waitObject)
 
     public static partial class ThreadPool
     {
-        // TODO: this is a very primitive (temporary) implementation of Thread Pool to allow Tasks to be
-        // used on Unix. All of this code must be replaced with proper implementation.
-
-        /// <summary>
-        /// Max allowed number of threads in the thread pool. This is just arbitrary number
-        /// that is used to prevent unbounded creation of threads.
-        /// It should by high enough to provide sufficient number of thread pool workers
-        /// in case if some threads get blocked while running user code.
-        /// </summary>
-        private static readonly int MaxThreadCount = 4 * ThreadPoolGlobals.processorCount;
-
-        /// <summary>
-        /// Semaphore that is used to release waiting thread pool workers when new work becomes available.
-        /// </summary>
-        private static SemaphoreSlim s_semaphore = new SemaphoreSlim(0);
-
-        /// <summary>
-        /// Number of worker threads created by the thread pool.
-        /// </summary>
-        private static volatile int s_workerCount = 0;
-
         public static bool SetMaxThreads(int workerThreads, int completionPortThreads)
         {
-            // Not supported at present
-            return false;
+            return ClrThreadPool.SetMaxThreads(workerThreads);
         }
 
         public static void GetMaxThreads(out int workerThreads, out int completionPortThreads)
         {
-            // Note that worker threads and completion port threads share the same thread pool.
-            // The total number of threads cannot exceed MaxThreadCount.
-            workerThreads = MaxThreadCount;
-            completionPortThreads = MaxThreadCount;
+            completionPortThreads = workerThreads = ClrThreadPool.GetMaxThreads();
         }
 
         public static bool SetMinThreads(int workerThreads, int completionPortThreads)
         {
-            // Not supported at present
-            return false;
+            return ClrThreadPool.SetMinThreads(workerThreads);
```

Comment:
I think it would be reasonable for SetMinThreads and SetMaxThreads to not allow setting a completion port thread count that is different from what GetMinThreads or GetMaxThreads always returns.